src/analyzer/analyzerwaveform.cpp

```diff
@@ -353,6 +351,15 @@ void AnalyzerWaveform::storeResults(TrackPointer tio) {
             m_waveform,
             m_waveformSummary);
 
+    // Set waveforms on track AFTER they'been written to disk in order to have
+    // a consistency when OverviewCache asks AnalysisDAO for a waveform summary.
+    if (m_waveform) {
```

Comment:
The old code removes the waveform from the track in case of nullptr. It hink that shall remain.